src/test/java/com/miquido/validoctor/rule/SimpleRuleTest.java

```diff
@@ -215,6 +215,16 @@ public void predefinedRule_each_collectionNotEmpty() {
     assertError(validoctor.examine(strings, collectionNotEmpty(), each(stringAlphabetic()), each(stringMinLength(2))));
   }
 
+  @Test
+  public void namedRule() {
+    String name = "custom_name";
+    Diagnosis diagnosis = validoctor.examine("a", "patient", named(name, stringMinLength(2)));
+    Set<Ailment> ailments = diagnosis.getAilments().get("patient");
+    assertTrue(ailments.stream().anyMatch(ailment -> ailment.getName().equals(name)));
+
+    assertOk(validoctor.examine("a", "patient", named(name, stringMaxLength(2))));
```

Comment:
assertOk does it give us anything?

Reply:
it checks if named didn't break anything if rule is fulfilled :D

Reply:
I don't think so. named() here is only a parameter. A parameter can't change a way the method behaves. And even if it could, this test is only one case of examine() method which doesn't gives us any guarantee.

Reply:
named() creates a new rule that is expected to be fulfilled or violated in exact same circumstances as the original one passed. This check aims to just generally make sure that rule predicate did not change when wrapping in named() (there is != usage in code there)